spec/controllers/moods_controller_spec.rb

```diff
@@ -0,0 +1,219 @@
+# frozen_string_literal: true
+
+RSpec.describe MoodsController, type: :controller do
+  let(:user) { create(:user1) }
+  let(:user_mood) { create(:mood, userid: user.id) }
+  let(:other_mood) { create(:mood, userid: user.id + 1) }
+  let(:valid_mood_params) { attributes_for(:mood).merge(userid: user.id) }
+  let(:invalid_mood_params) { { name: nil, description: nil } }
+
+  describe 'GET #index' do
+    context 'when the user is logged in' do
```

Comment:
Given the meta-programming involved in setting @moods, it's probably worth verifying it is set correctly and that @page_tooltip is set.